jobs/pacman-awsrules/src/main/java/com/tmobile/cloud/awsrules/federated/S3DPCEncryFederatedRule.java

```diff
@@ -0,0 +1,157 @@
+/*******************************************************************************
+ * Copyright 2019 T Mobile, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License.  You may obtain a copy
+ * of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ ******************************************************************************/
+/**
+  Copyright (C) 2019 T Mobile Inc - All Rights Reserve
+  Purpose:
+  Author :Avinash
+  Modified Date: Feb 27, 2019
+
+ **/
+package com.tmobile.cloud.awsrules.federated;
+
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.commons.lang.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.tmobile.cloud.awsrules.utils.PacmanUtils;
+import com.tmobile.cloud.constants.PacmanRuleConstants;
+import com.tmobile.pacman.commons.PacmanSdkConstants;
+import com.tmobile.pacman.commons.exception.InvalidInputException;
+import com.tmobile.pacman.commons.rule.BaseRule;
+import com.tmobile.pacman.commons.rule.PacmanRule;
+import com.tmobile.pacman.commons.rule.RuleResult;
+
+@PacmanRule(key = "check-for-s3-DPC-Encrypted-ACL", desc = "checks entirely for S3 Buckets With Global Write Permission", severity = PacmanSdkConstants.SEV_HIGH, category = PacmanSdkConstants.SECURITY)
+public class S3DPCEncryFederatedRule extends BaseRule {
+
+	private static final Logger logger = LoggerFactory.getLogger(S3DPCEncryFederatedRule.class);
+
+	/**
+	 * The method will get triggered from Rule Engine with following parameters
+	 * 
+	 * @param ruleParam
+	 * 
+	 *            ************* Following are the Rule Parameters********* <br><br>
+	 * 
+	 *            apiKeyName : Value of API key <br><br>
+	 * 
+	 *            apiKeyValue : Value of the API key name <br><br>
+	 * 
+	 *            apiGWURL : API gateway URL <br><br>
+	 * 
+	 *            ruleKey : check-for-s3-global-write-access <br><br>
+	 *            severity : Enter the value of severity <br><br>
+	 * 
+	 *            ruleCategory : Enter the value of category <br><br>
+	 *            
+	 *            roleIdentifyingString : Configure it as role/pac_ro <br><br>
```

Comment:
This is needed only when if your creating the client and describing.So you can remove it

Reply:
All the review changes are incorporated.